conformance/utils/kubernetes/helper.go

```diff
@@ -71,45 +76,28 @@ func PokeServer(t *testing.T, clientNamespace, clientPod, protocol, targetHost s
 
 // NamespacesMustBeReady waits until all Pods are marked Ready. This will
 // cause the test to halt if the specified timeout is exceeded.
-func NamespacesMustBeReady(t *testing.T, c client.Client, timeoutConfig config.TimeoutConfig, namespaces []string) {
+func NamespacesMustBeReady(t *testing.T, c client.Client, timeoutConfig config.TimeoutConfig, namespaces []string, statefulSetNames []string) {
 	t.Helper()
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.NamespacesMustBeReady)
+	defer cancel()
 
-	waitErr := wait.PollImmediate(1*time.Second, timeoutConfig.NamespacesMustBeReady, func() (bool, error) {
-		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-		defer cancel()
-
-		for _, ns := range namespaces {
-			podList := &v1.PodList{}
-			err := c.List(ctx, podList, client.InNamespace(ns))
-			if err != nil {
-				t.Errorf("Error listing Pods: %v", err)
+	waitErr := wait.PollUntilContextTimeout(ctx, 1*time.Second, timeoutConfig.NamespacesMustBeReady, true, func(ctx context.Context) (bool, error) {
+		for i, ns := range namespaces {
+			statefulSet := &appsv1.StatefulSet{}
+			statefulSetKey := types.NamespacedName{
+				Namespace: ns,
+				Name:      statefulSetNames[i],
 			}
-			for _, pod := range podList.Items {
-				if !findPodConditionInList(t, pod.Status.Conditions, "Ready", "True") &&
-					pod.Status.Phase != v1.PodSucceeded {
-					t.Logf("%s/%s Pod not ready yet", ns, pod.Name)
-					return false, nil
-				}
+			if err := c.Get(ctx, statefulSetKey, statefulSet); err != nil {
+				t.Errorf("Error retrieving StatefulSet %s from namespace %s: %v", statefulSetNames[i], ns, err)
+			}
+			if statefulSet.Status.ReadyReplicas != numStatufulSetReplicas {
```

Comment:
do you think we should also check for podIPs to be assigned ? It's possible for the number to be ok but maybe the containers are not fully ready yet? Either that or we add some buffer sleep before return to ensure state of all containers are in running state...

Reply:
or another idea is once the stateful set number check is done, we still keep the pod check as well ? that way list won't be empty and pod running is still considered?

Reply:
I think it's fine; ReadyReplicas should already capture the same info. I've tested it and it works as expected:
```
    suite.go:119: Test Setup: Ensuring Namespaces and Pods from base manifests are ready
    helper.go:95: StatefulSet replicas in namespace network-policy-conformance-gryffindor not rolled out yet. 0/2 replicas are availble.
    helper.go:95: StatefulSet replicas in namespace network-policy-conformance-gryffindor not rolled out yet. 0/2 replicas are availble.
    helper.go:95: StatefulSet replicas in namespace network-policy-conformance-gryffindor not rolled out yet. 0/2 replicas are availble.
    helper.go:95: StatefulSet replicas in namespace network-policy-conformance-gryffindor not rolled out yet. 1/2 replicas are availble.
    helper.go:95: StatefulSet replicas in namespace network-policy-conformance-gryffindor not rolled out yet. 1/2 replicas are availble.
    helper.go:99: Namespaces and Pods in network-policy-conformance-gryffindor, network-policy-conformance-slytherin, network-policy-conformance-hufflepuff, network-policy-conformance-ravenclaw namespaces are ready
```

Reply:
ack trusting you on this one then, lgtm